app/src/main/java/org/oppia/android/app/home/HomeViewModel.kt

```diff
@@ -107,20 +115,37 @@ class HomeViewModel(
   }
 
   /**
-   * Returns a [HomeItemViewModel] corresponding to the promoted stories to be displayed for this learner
-   * (see [PromotedStoryListViewModel]), or null if this profile does not have any promoted stories.
-   * Promoted stories are determined by any recent stories started by this profile.
+   * Returns a [HomeItemViewModel] corresponding to the promoted stories[PromotedStoryListViewModel] and Upcoming topics
+   * [ComingSoonTopicListViewModel]  to be displayed for this learner or null if this profile does not have any promoted stories.
+   * Promoted stories are determined by any recent stories or suggested stories started by this profile.
    */
-  private fun computePromotedStoryListViewModel(
-    ongoingStoryList: OngoingStoryList
+  private fun computeRecommendedActivityListViewModel(
```

Comment:
This method is called from 
```kotlin
listOfNotNull(
        computeWelcomeViewModel(profile),
        computeRecommendedActivityListViewModel(recommendedActivityList)
      )
```

but null is possible

Reply:
listOfNotNull function ignores null objects if they are added. So even it computeRecommendedActivityListViewModel() returns null it handles without throwing any error.